### PR TITLE
Topic Recognition Project 5 (s4642941)

### PR DESCRIPTION
**Pull request by rory2610**
Author: Rory Macleod - s4642941

This pull request includes an initial implementation of a Global Filter Network Model to classify Alzheimer's Diseased brains using MRI brain scans from the ADNI dataset.

The branch includes the following files:
- train.py : for training, validating and testing the model
- predict.py : for an example usage of the trained model
- dataset.py : for loading and augmenting the ADNI dataset
- modules.py : for GFNet model and supporting modules
- README.md : Documentation of the project

Additionally an image folder and files are included for the README.md file.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm can solve the problem to some extent. 
- Accuracy in testing dataset: 0.52. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation. However, applying a centre crop after resizing with the same shape makes no sense, so the only two data augmentation methods are random erasing and normalisation with mean and std 0.25 and 0.25, respectively.
- No validation observation: The validation dataset in the code is test dataset in fact, which does not match the description in the report - train, test, and validation.

**Code design:** Good.

**Code comment and docstring:**

- Minimal code comments 
- No function docstrings
- Some comment off parts are unnecessary and appear disorganized.

**Difficulty:** Hard.

**Additional Comments:**

- intermediate numbers of commits
- The model training result shows overfitting.
- There should be a conclusion in the report.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
Descriptive enough about the File Structure.
The pull request shows no incorporation of previous feedback .


File Organizing: Well-organized files.

Commit Log:
Commit messages are meaningful .
Commits demonstrate a logical development flow.
Commits are regularly made.

Documentation:
The README file provides a clear explanation of the algorithm and the problem it solves.
Code comments and docstrings are not  included .
Proper GitHub markdown formatting is used but could be worked upon for more readability . 
Can add about Conclusion and future works  of the report .

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Acc low. No validation. Overfillting visible.</td><td>-3</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. No doc strings. Unwanted content.</td><td>-1</td></tr>
	<tr><td></td><td>Data leakage found. Uses Data Aug, but has problems.</td><td>-1</td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. </td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-10</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Cant merge because of conflicting changes to main repo files (gitignore). Please update for merge, doesn't affect grade.
